### ISSUE TITLE
[Feature requests/ideas] Weapon Fire & Sound FX

### ISSUE THREAD

**Issue by BLOODWOLF333** [MegaMek/megamek#256]
I dont know how best to implement the animations but, for sound effects I just imagine MM playing the sound when the player clicks fire on a weapon.

Small Laser w/Light Hit (1-4 damage)
![smalllaser](https://cloud.githubusercontent.com/assets/16950832/16104780/d52ae222-3349-11e6-89c2-9f87f05de100.gif)
Medium Laser w/Medium Hit (5-9 damage)
![mediumlaser](https://cloud.githubusercontent.com/assets/16950832/16104793/ef9ab092-3349-11e6-9f9a-633a40783918.gif)
Large Laser w/Heavy Hit (10-14 damage)
![largelaser](https://cloud.githubusercontent.com/assets/16950832/16104796/f91c7e66-3349-11e6-83bf-9865b2d8476b.gif)
PPC w/Heavy Hit
![ppc](https://cloud.githubusercontent.com/assets/16950832/16104799/fce9c300-3349-11e6-9c48-c674db942530.gif)
Flamer
![flamer](https://cloud.githubusercontent.com/assets/16950832/16104800/00ba96a8-334a-11e6-84a4-c1ad4537f7b1.gif)
AC2 & Machine Gun
![ac2_mg](https://cloud.githubusercontent.com/assets/16950832/16104804/0428cf9e-334a-11e6-8662-ac9ee642e2bd.gif)
AC5
![ac5](https://cloud.githubusercontent.com/assets/16950832/16104806/0759c70e-334a-11e6-8dfb-b1b960298dea.gif)
AC10
![ac10](https://cloud.githubusercontent.com/assets/16950832/16104810/0a83b980-334a-11e6-8206-3e2423246dbb.gif)
AC20 w/VeryHeavy Hit (15+ damage)
![ac20](https://cloud.githubusercontent.com/assets/16950832/16104811/0dcf430c-334a-11e6-82c5-9fe9dc834cab.gif)
Gauss
![gauss](https://cloud.githubusercontent.com/assets/16950832/16104814/118a5734-334a-11e6-9ee8-25891f1c2b2c.gif)
LRM w/Ammo Explosion
![lrmammo](https://cloud.githubusercontent.com/assets/16950832/16104816/169f8c58-334a-11e6-8179-47fc0851270a.gif)
SRM w/Engine Explosion
![srmengine](https://cloud.githubusercontent.com/assets/16950832/16104819/1bfa70dc-334a-11e6-93fb-4a5b74d678f2.gif)

[SFX.zip](https://github.com/MegaMek/megamek/files/317638/SFX.zip)

**Comment by pheonixstorm** [comment]
Would be neat but I would think there would need to be time between the end of the firing phase and physical phase to display it unless it should display when the weapon is fired which would really not be worth it as each shot is fired. Would be much better as a feature to be displayed after all weapons on a unit have been fired.

**Comment by BLOODWOLF333** [comment]
Yes, that is what Arlith came up with last I talked to him about it.  Just a new phase in between that players would have the option of shutting off if they dont want to deal with it or would take too much time (company + games).  And I think I forgot to label it but, flamer would probably use the explosion sound effect.  Also made physical hit/collision images too in the zip (black and grey confetti like in the wreck images for each unit).

**Comment by HammerGS** [comment]
My two bits.  Like it. A lot.

**Comment by pheonixstorm** [comment]
With the AC hits I would hope that they would be variable/scriptable since fluff wise std ACs fire multiple rounds.

But even so you didn't sow anything for LB, Ultra, or Rotary. Those in the zip BW or were they forgotten/overlooked?

**Comment by BLOODWOLF333** [comment]
I just did the intro tech level weapons + gauss.  Are you talking about making different _shot images for the various ACs?  Rapid fire/ultras I would just think the animation could play twice but, I didnt get any indicated that this would be put in anytime soon so, didnt really go all out.

**Comment by pheonixstorm** [comment]
Byond the Ultra/Rotary if you read the fluff on the autocannons it talks about ACs firing various bursts. Like an Ac/20 firing a burst of X number of shells of different sizes which in TT rules are considered a single shot. So translated into an image you could have a tight group of say five shells together while for Ultra or Rotary each shot could be in groups placed farther apart ie 1 group of 5 followed farther behind by another group of 5. Program wise this could be a fluff option to use the group shots. LB weapons firing cluster could have a shotgun style animation of multiple pellets.

Overall wasn't aware of how much  you put into this beyond the above examples which is why I asked about the additional imagery. I do hope this makes it in at some point though, I think it would look really good compared to all the attack arrows. SFX I wouldn't use, but the additional GFX I know I would.

**Comment by BLOODWOLF333** [comment]
If I made the normal AC shots multiple rounds like in the fluff I worry that would make someone put in a bug report asking if ACs should be cluster hit weapons?  It is no problem to make the additional images for the more advanced weapons and will if this gets a green light.

**Comment by pheonixstorm** [comment]
The images would be for the fluff an optioned accordingly, so if someone thought it bugged and had the fluff option checked they would be told such and the ticket written off as invalid

**Comment by HammerGS** [comment]
Some suggestions

- I'd suggest for Cluster rounds to use a series of Dots.
- For Missiles I would try to show different numbers of missiles to cover the different size of racks
- Maybe its just me, but they describe PPCs as man made lighting.
- I would set up stream of rounds affect for non-cluster AC rounds
- I would create two different hit animations (One for normal hits and a larger/explosion one to reflect the crit regardless of what the crit is)

**Comment by MGM175** [comment]
Dammit... it's awesome :D
What about implementing this as a server-side option?

**Comment by BLOODWOLF333** [comment]
Example of an Atlas firing
https://www.youtube.com/watch?v=8WgLHTzCYpU&feature=youtu.be
